### DISCUSSION TITLE
GitHub Build Failing with: The following actions uses node12 which is deprecated

### DISCUSSION THREAD

**Discussion by valhuber** [squidfunk/mkdocs-material#6237] [Q&A] [answered]
### Bug description

Pushing a minor change to my docs resulted in the following GitHub build error from the doc auto-build:

```log
[deploy](https://github.com/ApiLogicServer/Docs/actions/runs/6607804942/job/17945648585)
The following actions uses node12 which is deprecated and will be forced to run on node16: actions/cache@v2. For more info: https://github.blog/changelog/2023-06-13-github-actions-all-actions-will-run-on-node16-instead-of-node12-by-default/
```

When I look up the GitHub link, it says to update this, but there is no such entry in my workflow:
```yaml
runs:
  using: 'node20'
  main: 'main.js'
```

Various stackoverflow entries suggested updating the requirements.txt, which I believe is now current.

My Doc Site is: [https://github.com/ApiLogicServer/Docs](https://github.com/ApiLogicServer/Docs).

Here is the [workflow](https://github.com/ApiLogicServer/Docs/blob/main/.github/workflows/ApiLogicServer.yml), which was built by following the doc'd procedures (years ago, don't recall where I got it).

I am happy to open the repo so you can issue a build if that proves necessary.

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
-

### Reproduction

This happens on a push into the docs repp.

### Steps to reproduce

1. Update an md file and push 
2. Observe failure (images attached)
<img width="1377" alt="job-failed" src="https://github.com/squidfunk/mkdocs-material/assets/5579631/21ae0df3-6b7c-4173-980f-58fd1586da11">

<img width="1377" alt="workflow-file" src="https://github.com/squidfunk/mkdocs-material/assets/5579631/b196ceca-f9af-44b0-857a-f0eff290854b">

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting. I'm not sure where this is coming from, but we are not advertising use of Node.js in any of our guides. I'm converting this to a discussion, since this seems to be a general problem. Additionally, no reproduction was provided, so nothing we can look into.

**Reply by squidfunk** [reply, marked as answer]
Well, as the error message says, you need to update `actions/cache`:

``` yaml
uses: actions/cache@v3
```

**Comment by valhuber** [comment]
Made that change.  Now fails with "exit 1"

<img width="1377" alt="workflow-file-changed" src="https://github.com/squidfunk/mkdocs-material/assets/5579631/0eea0375-cc2e-42dc-8ee5-929ee3e5ac8a">

<img width="1377" alt="fails-exit code 1" src="https://github.com/squidfunk/mkdocs-material/assets/5579631/27a51cf9-fd69-49df-946b-ae56d93d4867">

**Reply by squidfunk** [reply]
Please read the stack trace... it says that there are unrecognized links and you're building in strict mode, so any warning fails the build. If you fix the links, the build will succeed. I recommend doing this locally, much easier than going through CI every time.

**Comment by valhuber** [comment]
All set.

I use local all the time.  I did not realize it was failing due to (in my scramble) I had updated from medics 1.4. to 1.5 (the change to the link format).
